kernel/src/lib.rs

```diff
@@ -121,7 +121,7 @@ pub struct MemoryBlock {
 pub fn num_pages(nbytes: u64) -> usize {
     let npages = nbytes / PAGE_SIZE as u64;
     let remainder = nbytes % PAGE_SIZE as u64;
-    if remainder != 0 {
+    if remainder != 0 || nbytes < PAGE_SIZE as u64 {
```

Comment:
We may be able to drop `nbytes < PAGE_SIZE as u64` here: [`16 % 65536` should be `16`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f5a8105670e917110d04a38d089a988b), so the `remainder != 0` check _should_ handle the case where we're requesting fewer bytes than a full page.

Reply:
Yeah, this doesn't seem to be the issue. I will close it and include the other change in another PR.